### DISCUSSION TITLE
Point-to-Point routing alternatives

### DISCUSSION THREAD

**Discussion by Ciutnik** [devemux86/cruiser#234] [Q&A] [answered]
When planning short trips "there and back", I set several via-points to visit and let BRouter calculate alternatives, from which I choose the most interesting one. This works great, but we don't have a choice of the course of individual sections between points, but only the entire route. 

Is it possible to change the behavior of BRouter so that individual sections are as if individual routes, with the possibility of choosing alternatives? Such a possibility would make it much easier, for example, to choose the best of the alternative return routes, along previously unvisited roads. 

This could be a precursor to the future function of selecting route parameters for individual sections 😉

**Comment by devemux86** [comment, marked as answer]
@Ciutnik thanks for the very interesting idea!

BRouter (or any other router) does not seem to offer such a feature.
It would require processing from the app to handle multiple routings.

Something like this is already planned, please see here:
- https://github.com/devemux86/cruiser/discussions/142

**Reply by Ciutnik** [reply]
Thank you and waiting patiently for new features 👍

**Reply by devemux86** [reply]
@Ciutnik let's mark this as answered and continue the discussion in the [existing topic](https://github.com/devemux86/cruiser/discussions/142) to keep everything together.

I also copied your idea there:
- https://github.com/devemux86/cruiser/discussions/142#discussioncomment-6360176

**Comment by Cer-o** [comment]
Are you referring to the kurviger style in which you can choose a different type of route between waypoints?
Something like that would be Cruiser's crowning glory.

**Reply by Ciutnik** [reply]
We've already promised something similar! 👍

I am currently thinking about *alternative routes* so that their choice does not apply to the entire route, but to individual sections. 
Difficult to implement, but perhaps related to the sections known from K2.

**Reply by devemux86** [reply]
@Cer-o

It will be more advanced, as there will be no need to calculate the whole route with each change.

An advanced smart route calculation already exists in Cruiser navigation in the partial (re)routing.
Where Cruiser only calculates route until next waypoint and _your curvy routes remain unchanged_.

The plan is to offer such feature with more options in route planning. :slightly_smiling_face: 
- https://github.com/devemux86/cruiser/discussions/142
